ophyd/controls/device.py

```diff
@@ -489,10 +490,22 @@ def configure(self, d=None):
 
         Returns
         -------
-        (old, new) tuple
+        (old, new) tuple of dictionaries
         Where old and new are pre- and post-configure configuration states.
         '''
-        return {}, {}
+        old = self.read_configuration()
+        for key, val in d.items():
+            if key not in self.configuration_attrs:
+                # a little extra checking for a more specific error msg
+                if key not in self.signal_names:
+                    raise ValueError("there is no signal named %s", key)
+                else:
+                    raise ValueError("%s is not one of the "
+                                     "configuration_fields, so it cannot be "
+                                     "changed using configure", key)
+            getattr(self, key).put(val)
```

Comment:
@tacaswell serial, not parallel, puts should be the default here (with reference to waiting for them at the end). Underlying AD stuff doesn't usually handle rapid reconfiguration very well.